types/utils.go

```diff
@@ -250,6 +250,12 @@ func GetNodeKey(nodeKey *p2p.NodeKey) (crypto.PrivKey, error) {
 			return nil, fmt.Errorf("error unmarshalling node private key: %w", err)
 		}
 		return privKey, nil
+	case "secp256k1":
```

Comment:
[non-blocking] Should we use the same type method when we are checking for the key type? 

Here we are using the explicit `.Type()` method where in the manager sign method we are using `.(type)` casting.

Is there are reason we can't always use the `Type()` method?

Reply:
fixed..